### DISCUSSION TITLE
v0.99.5 Release Notes

### DISCUSSION THREAD

**Discussion by endigo9740** [skeletonlabs/skeleton#872] [Announcements]
# News

### Skeleton on the LogRock Podcast

Chris will be appearing on an episode of the [LogRocket Podcast](https://blog.logrocket.com/) on February 15th. Be sure to tune in as we'll be sharing some exciting news about Skeleton!

- [Where to Subscribe](https://podrocket.logrocket.com/subscribe)
- [View the full announcement](https://github.com/skeletonlabs/skeleton/discussions/836)

### Community Updates

We've introduced a new set of community focused updates to help ensure that all Skeleton projects are friendly and inclusive. These rules and guidelines will be enforced on Github, Discord, and any other community focused space.

- [View the new Code of Conduct](https://github.com/skeletonlabs/community/blob/main/CODE_OF_CONDUCT.md)
- [View the full announcement](https://github.com/skeletonlabs/skeleton/discussions/818)

# Release Notes

Release: https://github.com/skeletonlabs/skeleton/releases/tag/0.99.5

Today's release kicks off the refinement phase for the project as we move towards Release Candidate status for Skeleton. This means we'll be focusing primarily on polish and bugfixes. Please be aware that we expect the next few release to be bumpy, with more breaking changes than you may be used to. We ask for your patience during this time. However, be aware this will come with many highly requested updates and changes that will result in a much more stable product overall.

Summary:

- 🎨 New universal variant styles
- 🔘 Added button group styles
- 🔲 Added input group styles
- 📙 Greatly expanded Element documentation
- 🐞 Various bugfixes and minor improvements

## Tailwind Element Updates - https://github.com/skeletonlabs/skeleton/issues/837

Our refinement and polish for Skeleton has started with Tailwind Elements. These are broad and far reaching styles that affect many parts of Skeleton applications. We have [compiled a detailed list of changes here](https://github.com/skeletonlabs/skeleton/issues/837). The most notable improvements include:

### Universal Variants

![Screen Shot 2023-01-23 at 1 58 39 PM](https://user-images.githubusercontent.com/1509726/214137511-642f3987-02cd-4638-b17d-5671343f4f22.png)

Element variant styles now pull from a shared pool of styles and are universal. These are now available to buttons, badges, chips, cards, and other features. See the migration guide below for more info.

### Button Groups

![Screen Shot 2023-01-23 at 1 59 34 PM](https://user-images.githubusercontent.com/1509726/214137662-246db322-ac43-422d-a2fd-e2d93a1c1ba7.png)

Button groups have now been introduced, allowing you to group a set of related actions.

### Input Groups

![Screen Shot 2023-01-23 at 2 00 10 PM](https://user-images.githubusercontent.com/1509726/214137802-a04d6176-ef8f-4265-b51d-b645236d6e29.png)

Input groups have been introduced, allowing you to append icons and group multiple inputs.

### Greatly Expanded Documentation

![Screen Shot 2023-01-23 at 2 00 48 PM](https://user-images.githubusercontent.com/1509726/214137929-5fc9c573-ff5b-4728-8cd3-37025fe43046.png)

The documentation for each section has been greatly expanded, including many previously undocumented features.

### Less Opinionated Styles

![Screen Shot 2023-01-23 at 2 01 34 PM](https://user-images.githubusercontent.com/1509726/214138078-60c78907-a84c-4924-86f3-0de928022f7f.png)

Styling for certain elements, such as links and button animations, are now less opinionated by default. Providing a simpler starting point for customization in your own projects.

### Presentation Features

![Screen Shot 2023-01-23 at 2 02 16 PM](https://user-images.githubusercontent.com/1509726/214138190-75450291-c65a-49ce-941e-ebba1a3f689b.png)

We've also introduced a new Tailwind > Presentation category, for features such as Gradient Headings that do not require Skeleton.

### ⚠️ Migration Guide

Given the scope of these changes, we would highly recommend [viewing this ticket](https://github.com/skeletonlabs/skeleton/issues/837) and review items marked `BREAKING:`. We also advise reviewing the latest documentation for each feature. The change to universal variants is the most broad so we've provided instruction below.

**Migrating to Universal Variants**

All elements will now pull from a universal pool of shared variant styles, rather than having a unique set per Element. These are much simpler to use in practice. For example:

- Was `<button class="btn btn-filled-primary">`, now `<button class="btn variant-filled-primary">`
- Was `<span class="badge badge-ghost-secondary">`, now `<span class="badge variant-ghost-secondary">`
- Was `<span class="chip chip-ringed-tertiary">`, now `<span class="chip variant-ringed-tertiary">`
- Was `<aside class="alert alert-primary">`, now `<aside class="alert variant-ghost-warning">`

Even elements that previously required passing color values can now utilize variants instead:

- Was `<div class="card !bg-error-500">`, now `<div class="card variant-filled-error">`

We recommend using your code editor's find/replace feature to update each instance. See a few examples below:

- Find: `btn-filled`, replace with: `variant-filled`
- Find: `badge-filled`, replace with: `variant-filled`
- Find: `chip-filled`, replace with: `variant-filled`

## Tailwind Plugin Change

We've [modified](https://github.com/skeletonlabs/skeleton/pull/841) how portions of the Skeleton Tailwind plugin are generated. It is [now more modular](https://github.com/skeletonlabs/skeleton/pull/859). Due to this change there is one minor breaking change introduced that you will need to address in existing projects. See the migration guide below.

### ⚠️ Migration Guide

Open your `tailwind.config.cjs` and prepend `...` before the plugin and `()` after the plugin:

```js
const config = {
  // ...
  plugins: [
    // Keep any existing plugins present and append the following:
    // 👇 ADD THIS! "..."
    ...require("@skeletonlabs/skeleton/tailwind/skeleton.cjs")(), // <--- ADD THIS!
  ],
};
```

## Minor Improvements

- As announced in the [last release](https://github.com/skeletonlabs/skeleton/discussions/786), Alerts, Breadcrumbs, and Divider components have been removed:
  https://github.com/skeletonlabs/skeleton/pull/831

### Bugfixes

- Resolved an issue preventing the use of most Toast presets:
  https://github.com/skeletonlabs/skeleton/pull/855
- Auto-generated Intellisense will now be limited to just classes:
  https://github.com/skeletonlabs/skeleton/pull/828
- File Button components now respect the `disabled` attribute:
  https://github.com/skeletonlabs/skeleton/pull/823
- Minor bugfix for Gradient Heading styles
  https://github.com/skeletonlabs/skeleton/pull/812
- Progress Radial text will now accept `0%`:
  https://github.com/skeletonlabs/skeleton/pull/843
- Fixed two minor issue related to Listboxes:
  https://github.com/skeletonlabs/skeleton/issues/845
  https://github.com/skeletonlabs/skeleton/issues/846

### Documentation

- Resolved an issue with the Table of Contents causing site navigation to become unresponsive:
  https://github.com/skeletonlabs/skeleton/pull/861
- Improved our onboarding experience, especially for users migrating from an existing project:
  https://github.com/skeletonlabs/skeleton/pull/871
- Fixed a minor issue related to theme selection:
  https://github.com/skeletonlabs/skeleton/pull/853
- We've changed from Google Analytics to Vercel Analytics, which we feel will be less invasive of privacy:
  https://github.com/skeletonlabs/skeleton/pull/867
- We've added @LukeHagar as our first major sponsor on the homepage:
  https://github.com/skeletonlabs/skeleton/issues/816
- Fixed various typos and errors:
  https://github.com/skeletonlabs/skeleton/issues/808
  https://github.com/skeletonlabs/skeleton/pull/838

# Preparing for Upcoming Releases

We do not typically discuss upcoming releases here, but given the amount of notable changes coming, we thought it might be helpful to explain what's coming down the pipeline over the next few releases. Expect to see a large amount of breaking issues as we move closer towards a stable Release Candidate status.

## Polish Passes

Per [this ticket](https://github.com/skeletonlabs/skeleton/issues/782), our current focus is polishing and refining as many core features within Skeleton as possible. The updates to Elements in this release are just the beginning. We have two other passes planned as follows:

1. https://github.com/skeletonlabs/skeleton/issues/857
2. Polish Pass: Utility features - coming soon ([put issues here](https://github.com/skeletonlabs/skeleton/issues/782))

We welcome your feedback for changes you would like to see for these features. Please provide these in linked tickets or reach out on Discord. Keep in mind that these will be polish/bugfix focused. Total refactors are not likely, so please keep your expectations reasonable.

## Documentation Updates

Per [this ticket](https://github.com/skeletonlabs/skeleton/issues/783), we have a large number of improvements for the Skeleton documentation website planned. This includes everything from design updates, improved text copy and instruction, to administrative tasks such as SEO.

If you're a skilled copywriter we could use your help! The plan is to take all volunteers, then divide and conquer, addressing issues from top to bottom - this includes fixing typos, grammar, and ensuring content is relevant and accurate. If you're available to help in this process then please reach out and let us know!

# Support Skeleton

Like what you see here? Wish to sponsor Skeleton directly? Consider supporting us via one of the following services:

- [Ko-Fi](https://ko-fi.com/skeletonlabs)
- [GitHub Sponsors](https://github.com/sponsors/skeletonlabs)
- [Patreon](https://www.patreon.com/user?u=83786276)

**Comment by VincentWo** [comment]
Is input-cell now a required class on input elements to be styled correctly (with border-radius etc.)?
i can't find it documented anywhere, but without it it does not work

**Reply by endigo9740** [reply]
Nope, no class is required for input elements. They should operate just the same as before. Just make sure you've installed the Tailwind forms plugin (same as always) and ensure you're using `all.css` which will come with the form styles.

If you're having issues feel free to post a ticket or reach out on Discord and we'll help you further.

**Comment by lgs** [comment]
that is not workin for me

**Reply by endigo9740** [reply]
@lgs you'll have to be a bit more specific. The release notes cover a variety of topics and changes.
